pages/src/App.js

```diff
@@ -11,39 +11,54 @@ class App extends Component {
   }
 
   componentWillMount () {
-    fetch('/todos').then(res =>
-      res.json().then(todos => {
-        this.setState({
-          todos: new Map().withMutations(map => {
+
+    // Create WebSocket connection.
+    let socket = new WebSocket('ws://localhost:3000');
+    this.socket = socket
+
+    // Receiving a message from the server
+    socket.onmessage = msg => {
+      let todos = JSON.parse(msg.data);
+      let {id} = todos
+      console.log(todos);
+      this.setState({ todos: Map(todos)})
+      /*
+      if(Object.keys(todos).length>0){
+      this.setState({ todos: new Map().withMutations(map => {
             todos.forEach(todo => {
               map.set(todo.id, todo)
             })
           })
-        })
-      })
-    )
-  }
+        })}*/
+      console.log("todos");
+    }
 
-  _onCreateTodo = todo => {
-    const { id } = todo
-    const { todos } = this.state
-    const previous = todos.get(id)
+    // Connection opened
+    socket.onopen = () => {
+      let msg = {data : "send me data"}
+      this.socket.send(JSON.stringify(msg));
+    }
```

Comment:
You do not need to send a message on connection, the server can simply gives you the list of existing todos on connection by itself.